src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperScopeManager.cs

```diff
@@ -26,23 +27,42 @@ public TagHelperScopeManager()
         /// <summary>
         /// Starts a <see cref="TagHelperExecutionContext"/> scope.
         /// </summary>
+        /// <param name="parentExecutionContext">The parent <see cref="TagHelperExecutionContext"/>.</param>
         /// <param name="tagName">The HTML tag name that the scope is associated with.</param>
         /// <param name="uniqueId">An identifier unique to the HTML element this scope is for.</param>
         /// <param name="executeChildContentAsync">A delegate used to execute the child content asynchronously.</param>
         /// <param name="startWritingScope">A delegate used to start a writing scope in a Razor page.</param>
         /// <param name="endWritingScope">A delegate used to end a writing scope in a Razor page.</param>
         /// <returns>A <see cref="TagHelperExecutionContext"/> to use.</returns>
-        public TagHelperExecutionContext Begin([NotNull] string tagName,
-                                               [NotNull] string uniqueId,
-                                               [NotNull] Func<Task> executeChildContentAsync,
-                                               [NotNull] Action startWritingScope,
-                                               [NotNull] Func<TextWriter> endWritingScope)
+        public TagHelperExecutionContext Begin(
+            TagHelperExecutionContext parentExecutionContext,
```

Comment:
shoot, why does this parameter exist?  doesn't `TagHelperScopeManager` already know the `parentExecutionContext` value e.g. it can calculate `_executionScopes.Count == 0 ? null : _executionScopes.Peek()`?